lua/tardis/controls/sh_control_teleport.lua

```diff
@@ -14,7 +14,7 @@ TARDIS:AddControl({
                 if result then
                     TARDIS:Message(ply, "Controls.Teleport.Demat")
                 else
-                    if self:GetData("doorstatereal", false) then
+                    if self:GetData("doorstatereal", false) and not TARDIS:GetSetting("teleport-door-autoclose", self) then
```

Comment:
This change itself is fine but it got me wondering why we're not just returning the reason for demat failure with another parameter returned from self:Demat and trying to work it out from the state. Maybe something for another issue/PR but thought I'd raise it

Reply:
https://github.com/MattJeanes/TARDIS/issues/851